### PR TITLE
feat: redux-persist opportunitiesSlice / opportunitiesApi migration

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR migrates the opportuniesSlice / opportunitiesApi slices persistence, so that we don't display Idle MATIC staking opportunities anymore, which are incompatible with our app given the fact that withdraws are made with an NFT, see: https://medium.com/idle-finance/expanding-risk-hedging-liquid-staking-stmatic-pyts-a6331e74b5ae

Currently untested, this is pending Idle update of their API endpoint so that MATIC opportunities aren't returned anymore.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/3407
- closes  https://github.com/shapeshift/web/issues/3434

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3439** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3439" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3439?utm_source=stack-comment).

File: src/state/store.ts [hunk 1]
```diff
@@ -12,16 +12,28 @@ import type { ReduxState } from './reducer'
...
 import { assetApi } from './slices/assetsSlice/assetsSlice'
 import { marketApi, marketData } from './slices/marketDataSlice/marketDataSlice'
-import { opportunitiesApi } from './slices/opportunitiesSlice/opportunitiesSlice'
+import { initialState, opportunitiesApi } from './slices/opportunitiesSlice/opportunitiesSlice'
 import { portfolioApi } from './slices/portfolioSlice/portfolioSlice'
 import * as selectors from './slices/selectors'
 import { txHistoryApi } from './slices/txHistorySlice/txHistorySlice'
 import { validatorDataApi } from './slices/validatorDataSlice/validatorDataSlice'
 
+const migrations = {
```
Comment (0xdef1cafe) [inline src/state/store.ts:21]: let's pull this out to a migration directory

Comment (0xdef1cafe) [reply src/state/store.ts:21]: then name them like `migration-000.ts`

Comment (0xdef1cafe) [reply src/state/store.ts:21]: we could do like `migration-000-clear-opportunities.ts` then dynamically apply them in order

**Review by 0xdef1cafe — APPROVED** [review]
noice

**Pull request merged**